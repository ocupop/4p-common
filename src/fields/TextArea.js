import React, { useState, useEffect } from 'react'
import Skeleton from 'react-loading-skeleton'
import Label from './label'

const TextArea = ({
  className, label, hint, rows, placeholder, type, field, form: { errors, touched },
}) => {
  const [loading, setLoading] = useState()
  const status = touched[field.name] && errors[field.name] ? 'is-invalid' : ''

  useEffect(() => {
    setLoading(true)

    setTimeout(() => {
      setLoading(false)
    }, 1000)
    return () => {}
  }, [])

  return (
    <div className={`form-group ${className}`}>
      <Label label={label} hint={hint} />
      {loading ? (
        <Skeleton height={77} />
      ) : (
        <textarea
          className={`form-control ${status}`}
          {...field}
          placeholder={placeholder}
          rows={rows}
          type={type}
        />
      )}
      {touched[field.name] && errors[field.name] && (
        <span className="invalid-feedback">{errors[field.name]}</span>
      )}
    </div>
  )
}

export default TextArea
