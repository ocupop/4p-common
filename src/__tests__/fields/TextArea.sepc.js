import React from "react";
import { create } from "react-test-renderer";
import TextArea from '../../fields/TextArea'

const mockFieldObject = {
  name: 'textArea',
  onChange: () => {

  },
  value: '',
}

const mockFormObject = {
  dirty: false,
  errors: {},
  touched: false,
}
describe("TextArea component", () => {
  test("Matches the snapshot", () => {
    const textArea = create(<TextArea
      className="textArea"
      hint="this is a hint"
      type="text"
      placeholder="foobar"
      required
      field={mockFieldObject}
      form={mockFormObject}
    />);
    expect(textArea.toJSON()).toMatchSnapshot();
  });
});
