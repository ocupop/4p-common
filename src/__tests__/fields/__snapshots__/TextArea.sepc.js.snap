// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="form-group textArea"
>
  <label
    className="sr-only"
  >
    <span
      className="mr-2"
    />
    <div
      className="pt-1 ml-1 d-inline-block"
    >
      <i
        className="ri-question-fill text-mid"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </label>
  <textarea
    className="form-control "
    name="textArea"
    onChange={[Function]}
    placeholder="foobar"
    type="text"
    value=""
  />
</div>
`;
